### PR TITLE
Created ViewModels and responseModel

### PR DESCRIPTION
**Pull request by VincentMoyo**
Created Development Pod for Crypto and Bitcoin ViewModel

### CODE REVIEW

File: CoolCurrencyDevelopmentPod/RepositoryModelFramework/CryptoAndMetalsRepositorys.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            task.resume()
+        }
+    }
+    
+    func URLBitcoinStringBuilder(for baseCurrency: String) -> URL? {
+        var urlComponents = URLComponents()
+        urlComponents.scheme = "https"
+        urlComponents.host = "rest.coinapi.io"
+        urlComponents.path = "/v1/exchangerate/BTC/" + baseCurrency
+        let apiKeyQueryItem = URLQueryItem(name: "apikey", value: "4645E475-133C-458A-AA48-2EB70A347301")
```
Comment (AbdullahNanaDVT) [inline CoolCurrencyDevelopmentPod/RepositoryModelFramework/CryptoAndMetalsRepositorys.swift:65]: You could use a Constants class for the API Key. Also, it shouldn't be committed.

File: CoolCurrencyDevelopmentPod/RepositoryModelFramework/CryptoAndMetalsRepositorys.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        return urlComponents.url
+    }
+    
+    func URLMetalsStringBuilder(for baseCurrency: String) -> URL? {
+        var urlComponents = URLComponents()
+        urlComponents.scheme = "https"
+        urlComponents.host =  "metals-api.com"
+        urlComponents.path = "/api/latest"
+        let apiKeyQueryItem = URLQueryItem(name: "access_key",
+                                           value: "lj7xeh4umkscma4cv84w4d67l26ic34hjo2r2ky7571560jv9b2zu331ta1p")
```
Comment (AbdullahNanaDVT) [inline CoolCurrencyDevelopmentPod/RepositoryModelFramework/CryptoAndMetalsRepositorys.swift:76]: Same as above

File: CoolCurrencyDevelopmentPod/RepositoryModelFramework/CryptoDatabaseRepository.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        self.database = databaseReference
+    }
+    
+    public func retrieveUserInformationFromDatabase(userID baseUser: String, completion: @escaping (Result<[String : String], Error>) -> Void) {
+        database.child("Users").child(baseUser).observeSingleEvent(of: .value) { snapshot in
+            guard let value = snapshot.value as? [String: String] else {
+                return
+            }
+            DispatchQueue.main.async {
+                print(value)
```
Comment (AbdullahNanaDVT) [inline CoolCurrencyDevelopmentPod/RepositoryModelFramework/CryptoDatabaseRepository.swift:25]: This can be deleted.

**Review by AbdullahNanaDVT — CHANGES_REQUESTED** [review]

File: CoolCurrencyDevelopmentPod/ViewModelFramework/Constant.swift [hunk 4]
```diff
@@ -0,0 +1,33 @@
+//
+//  Constants.swift
+//  CoolCurrencyDevelopmentPod
+//
+//  Created by Vincent Moyo on 2021/11/10.
+//
+
+import Foundation
+
+public struct Constant {
```
Comment (bgouws) [inline CoolCurrencyDevelopmentPod/ViewModelFramework/Constant.swift:10]: This will still be available on your local repo, please don't commit.

**Review by bgouws — CHANGES_REQUESTED** [review]

**Review by bgouws — APPROVED** [review]

**Pull request merged**